File: cloudmessanging4/app/src/main/java/com/example/cloudmessanging4/Signup.java
package com.example.cloudmessanging4;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.AppCompatButton;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.view.View;
import android.widget.EditText;
import android.widget.ProgressBar;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.gms.tasks.OnCompleteListener;
import com.google.android.gms.tasks.OnFailureListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.auth.AuthResult;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.FirebaseFirestore;

public class Signup extends AppCompatActivity {
    TextView loginclick;
    EditText name,lastname,email,mobile,password,cnfpassword;
    AppCompatButton Signup;
   private FirebaseAuth Fauth;
   FirebaseFirestore Fstore=FirebaseFirestore.getInstance();
ProgressBar progressBar;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_signup);
        loginclick=findViewById(R.id.textView6);
        name=findViewById(R.id.Firstname);
        lastname=findViewById(R.id.Lastname);
        email=findViewById(R.id.registeremail);
        mobile=findViewById(R.id.mobilenumber);
        password=findViewById(R.id.password);
        cnfpassword=findViewById(R.id.cnfpassword);
        Signup=findViewById(R.id.Signupbutton);
        progressBar=findViewById(R.id.progressbar);
        Fauth=FirebaseAuth.getInstance();

    Signup.setOnClickListener(new View.OnClickListener() {
        @Override
        public void onClick(View view) {
            String Name=name.getText().toString();
            String Lastname=lastname.getText().toString();
            String Email=email.getText().toString();
            String Number=mobile.getText().toString();
            String Password=password.getText().toString();
            String Cnfpassword=cnfpassword.getText().toString();

            if(Name.isEmpty()||Lastname.isEmpty()||Email.isEmpty()||Number.isEmpty()||Password.isEmpty()||Cnfpassword.isEmpty()){
                Toast.makeText(getApplicationContext(),"Enter all the field ",Toast.LENGTH_SHORT).show();
            }
            if(!Password.equals(Cnfpassword)){
                Toast.makeText(getApplicationContext(),"Passowrd Doesn't match",Toast.LENGTH_SHORT).show();

            }
            Fauth.createUserWithEmailAndPassword(Email,Password).addOnCompleteListener(new OnCompleteListener<AuthResult>() {
                @Override
                public void onComplete(@NonNull Task<AuthResult> task) {
                    if(task.isSuccessful()) {
//                        startActivity(new Intent(getApplicationContext(), Login.class));
                        Toast.makeText(getApplicationContext(), "Inserted", Toast.LENGTH_SHORT).show();

                        Fstore.collection("User").document(FirebaseAuth.getInstance().getUid()).set(new Contactadapter(Name,Lastname,Number,Email,Password,Cnfpassword));
                    }
                }
            }).addOnFailureListener(new OnFailureListener() {
                @Override
                public void onFailure(@NonNull Exception e) {
                    Toast.makeText(getApplicationContext(),e.getMessage(),Toast.LENGTH_SHORT).show();


                }
            });


        }
    });

//                String Name=name.getText().toString();
//                String Lastname=lastname.getText().toString();
//                String Email=email.getText().toString();
//                String Number=mobile.getText().toString();
//                String Password=password.getText().toString();
//                String Cnfpassword=cnfpassword.getText().toString();



                
               






        loginclick.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
//                startActivity(new Intent(getApplicationContext(),Login.class));
            }
        });
    }
}

File: Foodorderingappadminpanel/app/src/main/java/com/example/foodorderingappadminpanel/Signup.kt
package com.example.foodorderingappadminpanel

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import androidx.core.content.res.ResourcesCompat
import com.example.foodorderingappadminpanel.databinding.ActivitySignupBinding
import com.example.foodorderingappadminpanel.modelclass.signupmodelclass
import com.google.firebase.auth.ktx.auth
import com.google.firebase.database.ktx.database
import com.google.firebase.ktx.Firebase
import www.sanju.motiontoast.MotionToast
import www.sanju.motiontoast.MotionToastStyle

class Signup : AppCompatActivity() {
    private val binding by lazy {
        ActivitySignupBinding.inflate(layoutInflater)
    }
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(binding.root)
        binding.button.setOnClickListener {
            if(binding.Name.text.toString().equals("")||
                binding.Lastname.text.toString().equals("")||
                binding.Emailorphone.text.toString().equals("")||
                binding.Loginpassword.text.toString().equals("")||
                binding.cnfpassword.text.toString().equals("")){
                Toast.makeText(this, "Please fill the details", Toast.LENGTH_SHORT).show()
            }else if(!binding.Loginpassword.text.toString().equals(binding.cnfpassword.text.toString())){
                Toast.makeText(this, "password doesn't match", Toast.LENGTH_SHORT).show()



            }
            else{
                Firebase.auth.createUserWithEmailAndPassword(binding.Emailorphone.text.toString(),binding.Loginpassword.text.toString()).addOnCompleteListener {
                    if(it.isSuccessful) {

                        var signupmodelclass = signupmodelclass(
                            binding.Name.text.toString(),

                            binding.Lastname.text.toString(),
                            binding.Emailorphone.text.toString(),
                            binding.Loginpassword.text.toString()
                        )

                        Firebase.database.reference.child("Admin SignedUp User")
                            .child(Firebase.auth.currentUser!!.uid).setValue(signupmodelclass)
                            .addOnSuccessListener {
                                MotionToast.createToast(this,
                                    "Hurray success 😍",
                                    "Upload Completed successfully!",
                                    MotionToastStyle.SUCCESS,
                                    MotionToast.GRAVITY_BOTTOM,
                                    MotionToast.LONG_DURATION,
                                    ResourcesCompat.getFont(this, www.sanju.motiontoast.R.font.helveticabold))

//                                startActivity(Intent(this@Signup, LoginActivity::class.java))


                            }
                    }

                }.addOnFailureListener {
                    MotionToast.createToast(this,
                        "Failed ☹️",
                        "Profile Update Failed!",
                        MotionToastStyle.ERROR,
                        MotionToast.GRAVITY_BOTTOM,
                        MotionToast.LONG_DURATION,
                        ResourcesCompat.getFont(this, www.sanju.motiontoast.R.font.helveticabold))

                }
            }
        }












    }
}